src/main.rs

```diff
@@ -160,7 +161,7 @@ fn main() -> Result<(), Box<dyn std::error::Error>> {
         let client_server_map: ClientServerMap = Arc::new(Mutex::new(HashMap::new()));
 
         // Statistics reporting.
-        let (stats_tx, stats_rx) = mpsc::channel(100_000);
+        let (stats_tx, stats_rx) = mpsc::channel(500_000);
```

Comment:
great find! @levkk do you think we could setup an mpmc channel for this to avoid this bottleneck?

Reply:
Not sure how `mpmc` will help here, since we'll have to dedup the events in the end. The events count things like statistics, how many clients are connected, queries & transactions sent etc.

Current metrics implementation is a bit lax in guarantees of accuracy. I'm thinking more and more we should switch to atomics instead and count what we need to count on each `ConnectionPool`. Since atomics can be read and written to very quickly from many threads, I think it'll work nicely.

This change is good however, to keep us going. `500_000` should be enough, and the only constraint here is memory. The events themselves should be pretty small and since this is just a limit to help process burst workloads, I think this will continue to work.